qiskit/transpiler/coupling.py

```diff
@@ -320,6 +320,56 @@ def from_grid(cls, num_rows, num_columns, bidirectional=True):
         )
         return cmap
 
+    @classmethod
+    def from_heavy_hex(cls, distance, bidirectional=True):
+        """Return a heavy hexagon graph coupling map
+
+        A heavy hexagon graph is described in:
+
+        https://journals.aps.org/prx/abstract/10.1103/PhysRevX.10.011022
+
+        Args:
+            distance (int): The code distance for the generated heavy hex
+                graph. The value for distance can be any odd positive integer.
+                The distance relates to the number of qubits by:
+                :math:`n = \\frac{5d^2 - 2d - 1}{2}` where :math:`n` is the
+                number of qubits and :math:`d` is the ``distance`` parameter.
+            bidirectional (bool): Whether the edges in the output coupling
+                graph are bidirectional or not. By default this is set to
+                ``True``
```

Comment:
If `bidirectional=False`, what direction is chosen? I think this should be according to the paper too. See fig 10 in https://arxiv.org/pdf/1907.09528.pdf.

The black dots are control, and other colors are target. The edge direction in the graph should always go from control to target. This is for both heavy-hex and heavy-square.

<img width="707" alt="image" src="https://user-images.githubusercontent.com/8622381/134229635-802ab803-05ce-4506-aed7-7f663d9b869d.png">

Reply:
It does follow the diagram from the paper. This was something that I brought up in review on the retworkx PRs: https://github.com/Qiskit/retworkx/pull/293#pullrequestreview-732027160 and https://github.com/Qiskit/retworkx/pull/313#pullrequestreview-731994148